tests/integration-tests/chaos-experiments/chaos-experiments-context-test.js

```diff
@@ -0,0 +1,211 @@
+const should = require('should'),
+    uuid = require('uuid');
+
+const validHeaders = { 'Content-Type': 'application/json' };
+const chaosExperimentsRequestSender = require('./helpers/requestCreator');
+const testsRequestSender = require('../tests/helpers/requestCreator');
+const { ERROR_MESSAGES } = require('../../../src/common/consts');
+describe('Chaos experiments api - with contexts', function () {
+    let contextId;
+    this.timeout(5000000);
+    before(async function () {
+        contextId = uuid.v4().toString();
+        await chaosExperimentsRequestSender.init();
+        await testsRequestSender.init();
+    });
+
+    describe('Good requests', async function () {
+        let chaosExperimentResponse;
+        const chaosExperimentsInserted = [];
+        describe('GET /v1/chaos-experiments', function () {
+            before(async function () {
+                const headersWithRandomContext = Object.assign({}, validHeaders, { 'x-context-id': contextId });
+
+                for (let i = 0; i < 3; i++) {
+                    const chaosExperiment = generateRawChaosExperiment(uuid.v4(), contextId);
+                    chaosExperimentResponse = await chaosExperimentsRequestSender.createChaosExperiment(chaosExperiment, headersWithRandomContext);
+                    chaosExperimentsInserted.push(chaosExperimentResponse);
+                }
+            });
+
+            it('get chaos experiments with context_id should return all processors created with specific context', async function () {
+                const headers = {
+                    'Content-Type': 'application/json',
+                    'x-context-id': contextId
+                };
+                const getChaosExperimentsResponse = await chaosExperimentsRequestSender.getChaosExperiments(undefined, undefined, undefined, headers);
+
+                should(getChaosExperimentsResponse.statusCode).equal(200);
+                const chaosExperiments = getChaosExperimentsResponse.body;
+                should(chaosExperiments.length).equal(3);
+
+                const contextAResponse = getChaosExperimentsResponse.body.find(o => o.id === chaosExperimentResponse.body.id);
+                should(contextAResponse).not.be.undefined();
+            });
+
+            it('get chaos experiments with wrong context_id should no processors', async function () {
+                const headers = {
+                    'Content-Type': 'application/json',
+                    'x-context-id': uuid.v4()
```

Comment:
We must have context-id to use the API? Is is wanted?

Reply:
no...it's not required the predator allows it as like overall context